src/openms/source/ANALYSIS/OPENSWATH/DIAPrescoring.cpp

```diff
@@ -130,59 +131,85 @@ namespace OpenMS
   void DiaPrescore::score(OpenSwath::SpectrumPtr spec,
                           const std::vector<OpenSwath::LightTransition>& lt,
                           double& dotprod,
-                          double& manhattan)
+                          double& manhattan) const
   {
     std::vector<std::pair<double, double> > res;
-    getMZIntensityFromTransition(lt, res);
-    std::vector<double> firstIstotope, theomasses;
-    DIAHelpers::extractFirst(res, firstIstotope);
-    std::vector<std::pair<double, double> > spectrum, spectrum2;
-    DIAHelpers::addIsotopes2Spec(res, spectrum, nr_charges_);
```

Comment:
nr_charges_ here is at the parameter for nr_isotopes_, which is ok since the class here sets the same defaults. But this call does not set charge! Therefore this function was always called with charge=4 -> completely different isotope patterns since the mz is multiplied by 4 instead of usually 1.